crates/bevy_math/src/direction.rs

```diff
@@ -307,6 +336,43 @@ impl Dir3 {
     pub const fn as_vec3(&self) -> Vec3 {
         self.0
     }
+
+    /// Performs a spherical linear interpolation between `self` and `rhs`
+    /// based on the value `s`.
+    ///
+    /// This corresponds to interpolating between the two directions at a constant angular velocity.
+    ///
+    /// When `s == 0.0`, the result will be equal to `self`.
+    /// When `s == 1.0`, the result will be equal to `rhs`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_math::Dir3;
+    /// # use approx::RelativeEq;
+    /// #
+    /// let dir1 = Dir3::X;
+    /// let dir2 = Dir3::Y;
+    ///
+    /// let result1 = dir1.slerp(dir2, 1.0 / 3.0);
+    /// assert!(result1.relative_eq(&Dir3::from_xyz(0.75_f32.sqrt(), 0.5, 0.0).unwrap(), f32::EPSILON, f32::EPSILON));
+    ///
+    /// let result2 = dir1.slerp(dir2, 0.5);
+    /// assert!(result2.relative_eq(&Dir3::from_xyz(0.5_f32.sqrt(), 0.5_f32.sqrt(), 0.0).unwrap(), f32::EPSILON, f32::EPSILON));
+    /// ```
+    #[inline]
+    pub fn slerp(self, rhs: Self, s: f32) -> Self {
+        let mut axis = self.cross(rhs.0);
+        let axis_length_squared = axis.length_squared();
+        if axis_length_squared == 0.0 {
+            // Colinear directions have no rotation axis or angle between them, so the best is doable is lerp.
+            return Dir3(self.lerp(rhs.0, s));
+        }
+        axis /= axis_length_squared.sqrt();
+        let angle = self.angle_between(rhs.0);
+        let quat = Quat::from_axis_angle(axis, angle * s);
+        Dir3(quat.mul_vec3(self.0))
```

Comment:
This implementation should probably just use `Quat::from_rotation_arc` on the underlying vectors instead of trying to do all the logic itself.

Reply:
I changed to Quat calculation, and it is less precise
```
let quat = Quat::IDENTITY.slerp(
    Quat::from_rotation_arc(self.0, rhs.0),
    s,
);
Dir3(quat.mul_vec3(self.0))
```
gives error:
```
assert_relative_eq!(Dir3::X.slerp(Dir3::Z, 1.0 / 3.0), Dir3::from_xyz(0.75f32.sqrt(), 0.0, 0.5).unwrap())

    left  = Dir3(Vec3(0.86602515, 0.0, 0.49999994))
    right = Dir3(Vec3(0.8660254, 0.0, 0.5))
```

Reply:
I guess I don't really see that as a big problem; I'd still rather use `Quat::from_rotation_arc` than maintain additional logic which is essentially the same.